orchestrator-salt/src/main/resources/salt/salt/postgresql/scripts/backup_db.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ $# -ne 5 ] && [ $# -ne 6 ]; then
+  echo "Invalid inputs provided"
+  echo "Script accepts 5 inputs:"
+  echo "  1. Cloud Provider (azure | aws)"
+  echo "  2. Object Storage Service url to retrieve backups."
+  echo "  3. PostgreSQL host name."
+  echo "  4. PostgreSQL port."
+  echo "  5. PostgreSQL user name."
+  echo
+  echo "  Optional: PostgreSQL password."
+  exit 1
+fi
+
+CLOUD_PROVIDER="$1"
+BACKUP_LOCATION=$(echo "$2"| sed 's/\/\+$//g') # Clear trailng '/' (if present) for later path joining.
+HOST="$3"
+PORT="$4"
+USERNAME="$5"
+export PGPASSWORD="$6" # We can provide the password to pg_dump through this variable, or in ~/.pgpass
```

Comment:
I would not export the password as it will be part of the logs. In Salt we can simply pass passwords to the script without exposing it. Example: https://github.com/hortonworks/cloudbreak/blob/bb5ebdf54b0f987e232e381960cf6dc2d66ea8ed/freeipa/src/main/resources/freeipa-salt/salt/freeipa/common-install.sls#L5

Reply:
The problem actually lies in passing the password to the `pg_dump` command, there isn't any option to provide it as a command line argument -- if we stick to using a _password_ we need to either:
* Provide it interactively
* Put the password in the special `PGPASSWORD` variable
* Put the password as a connection string in `~/.pgpass`, and `chmod 0600` it for root.

Reply:
@keyki has explained that we can use Salt to define the `PGPASSWORD` variable in the environment, so there's no need to set it + export it in the backup or restore scripts.

This prevents accidental exposure of the password by `set -x`.